target/linux/brcm63xx/dts/agpwi.dts

```diff
@@ -0,0 +1,125 @@
+/dts-v1/;
+
+#include "bcm6328.dtsi"
+
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "ADB P.DG A4001N A-000-1A1-AX";
+	compatible = "adb,agpwi", "brcm,bcm6328";
+
+	chosen {
+		bootargs = "rootfstype=squashfs,jffs2 noinitrd console=ttyS0,115200";
+		stdout-path = "serial0:115200n8";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&pinctrl 23 1>;
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&pinctrl 24 1>;
+			linux,code = <KEY_WPS_BUTTON>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power_red {
+			label = "A4001N:red:power";
+			gpios = <&pinctrl 8 1>;
+		};
+		power_green {
+			label = "A4001N:green:power";
+			gpios = <&pinctrl 12 1>;
+			default-state = "on";
+		};
+		
+		adsl_red {
+			label = "A4001N:red:adsl";
+			gpios = <&pinctrl 5 1>;
+		};
+		adsl_green {
+			label = "A4001N:green:adsl";
+			gpios = <&pinctrl 3 1>;
+		};
+		
+		internet_red {
+			label = "A4001N:red:internet";
+			gpios = <&pinctrl 2 1>;
+		};
+		internet_green {
+			label = "A4001N:green:internet";
+			gpios = <&pinctrl 11 1>;
+		};
+		
+		wifi_red {
+			label = "A4001N:red:wifi";
+			gpios = <&pinctrl 10 1>;
+		};
+		wifi_green {
+			label = "A4001N:green:wifi";
+			gpios = <&pinctrl 9 1>;
+		};
+		
+		service_red {
+			label = "A4001N:red:service";
+			gpios = <&pinctrl 7 1>;
+		};
+		service_green {
+			label = "A4001N:green:service";
+			gpios = <&pinctrl 6 1>;
+		};
+	};
+};
+
+&hsspi {
+	status = "ok";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <16666667>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
+		reg = <0>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			cfe@0 {
+				reg = <0x000000 0x010000>;
+				label = "cfe";
+				read-only;
+			};
+
+			linux@10000 {
+				reg = <0x010000 0x7e0000>;
+				label = "linux";
+				compatible = "brcm,bcm963xx-imagetag";
+			};
+
+			nvram@7f0000 {
+				reg = <0x7f0000 0x010000>;
+				label = "nvram";
+			};
+		};
+	};
+};
+
+&uart0 {
+	status = "ok";
```

Comment:
I think "okay" is prefered here.

Reply:
I copied the content of this file from the A4001N.dts and just modified GPIO stuff, that's where this "ok" comes from. I left "ok" for code style uniformity with the tween PCB A4001N.

Do you steel want me to change "ok" into "okay"?

Reply:
okay is preferred, while ok also works. Since we add something new here, please update all ok to okay in the DTS.